### PR TITLE
Refactored DSA to support versions and tests pass

### PR DESCRIPTION
**Pull request by DakshMiglani**

### CODE REVIEW

File: src/dsa.ts [hunk 1]
```diff
@@ -133,7 +133,7 @@ export class DSA {
 
   public async getAcocuntIdDetails(instanceId: Instance['id']) {
     try {
-      const contract = new this.web3.eth.Contract(Abi.read.core, Addresses.read.core)
+      const contract = new this.web3.eth.Contract(Abi.read, Addresses.core.read)
```
Comment (thrilok209) [inline src/dsa.ts:136]: @DakshMiglani Let's use `Abi.read.core`?

Comment (DakshMiglani) [reply src/dsa.ts:136]: image

Abi.read.core doesn't exist with the new file structure from ba2dfa6

Comment (thrilok209) [reply src/dsa.ts:136]: Sorry! I meant `Abi.core.read`

Comment (DakshMiglani) [reply src/dsa.ts:136]: Yes, that's exactly what I used!

**Pull request merged**